simulator/modules/core/perf_sim.h

```diff
@@ -34,6 +34,7 @@ class PerfSim : public CycleAccurateSimulator
     Trap run( uint64 instrs_to_run) final;
     void set_target( const Target& target) final;
     void set_memory( std::shared_ptr<FuncMemory> memory) final;
+    void set_kernel( std::shared_ptr<Kernel> kernel) final { this->kernel = kernel; }
```

Comment:
We generally do not use `this` and avoid name shadowing. Please use other name for a local variable (`k` would be fine).